codersdk/workspaceagents.go

```diff
@@ -143,9 +143,17 @@ func (c *Client) DialWorkspaceAgent(ctx context.Context, agentID uuid.UUID, opti
 	}
 
 	ip := tailnet.IP()
+	var header http.Header
+	headerTransport, ok := c.HTTPClient.Transport.(interface {
+		Header() http.Header
+	})
+	if ok {
+		header = headerTransport.Header().Clone()
```

Comment:
(Accidentally deleted this comment): We seem to clone in both the function and here, not an issue per-se just pointing it out.

Reply:
Ahh, I'll change that!

Reply:
Fixed!